### PR TITLE
Fix crash on Linux on startup

### PR DESCRIPTION
**Pull request by veler**
## Pull request type

Please check the type of change your PR introduces:

- [X] Bugfix
- [ ] New feature or enhancement
- [ ] UI change (please include screenshot!)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Internationalization and localization
- [ ] Other (please describe):

## What is the current behavior?

In GTK 4.10 on Ubuntu, using either [GTKSharp](https://github.com/GtkSharp/GtkSharp) or [Gir.Core](https://gircore.github.io/) NuGet package, we're unable to invoke [gtk_file_dialog_new](https://docs.gtk.org/gtk4/ctor.FileDialog.new.html) to create a new file dialog. Why? I have no clue.

## What is the new behavior?

Workaround this issue by using GTK's [FileChooserNative](https://docs.gtk.org/gtk3/class.FileChooserNative.html) API.

## Other information

This PR also includes a small refactoring and more bug fixes:
- ThemeListener is now able to properly switch between the light and dark theme.
- Fixed a bug where we were trying to navigate to `wwwroot/index.html` from the wrong root directory. I think in the future we should consider using an embedded resource instead, if possible.
- Tiny refactoring in DevToys MacOS's BlazorWebView.

image

## Quality check

Before creating this PR:

- [X] Did you follow the code style guideline as described in [CONTRIBUTING.md](https://github.com/veler/DevToys/blob/main/CONTRIBUTING.md)
- [X] Did you build the app and test your changes?
- [ ] Did you check for accessibility? On Windows, you can use Accessibility Insights for this.
- [X] Did you verify that the change work in Release build configuration
- [ ] Did you verify that all unit tests pass
- [X] If necessary and if possible, did you verify your changes on:
   - [X] Windows
   - [X] macOS (DevToys 2.0)
   - [X] Linux (DevToys 2.0)

### CODE REVIEW

File: src/app/dev/platforms/desktop/DevToys.Linux/Components/BlazorWebView/BlazorWebView.cs [hunk 1]
```diff
@@ -7,6 +7,11 @@ internal sealed class BlazorWebView : WebView
 {
     private readonly BlazorWebViewBridge _bridge;
 
+    static BlazorWebView()
+    {
+        WebKit.Module.Initialize();
```
Comment (veler) [inline src/app/dev/platforms/desktop/DevToys.Linux/Components/BlazorWebView/BlazorWebView.cs:12]: Just delaying the loading of WebKit from `LinuxProgram.cs` to here.

File: src/app/dev/platforms/desktop/DevToys.Linux/Components/BlazorWebView/BlazorWebViewOptions.cs [hunk 2]
```diff
@@ -6,7 +8,7 @@ internal record BlazorWebViewOptions
 
     internal string HostPath { get; init; } = Path.Combine("wwwroot", "index.html");
 
-    internal string ContentRoot => Path.GetDirectoryName(Path.GetFullPath(HostPath))!;
+    internal string ContentRoot { get; init; } = Path.GetDirectoryName(Assembly.GetEntryAssembly()!.Location)!;
```
Comment (veler) [inline src/app/dev/platforms/desktop/DevToys.Linux/Components/BlazorWebView/BlazorWebViewOptions.cs:11]: This fix the bug I was talking about in the description where we were looking at `wwwroot/index.html` from the wrong root folder.

File: src/app/dev/platforms/desktop/DevToys.Linux/Core/FileStorage.cs [hunk 3]
```diff
@@ -6,16 +6,8 @@
 namespace DevToys.Linux.Core;
 
 [Export(typeof(IFileStorage))]
-internal sealed class FileStorage : GObject.Object, IFileStorage
+internal sealed class FileStorage : IFileStorage
 {
-    [ImportingConstructor]
-    public FileStorage()
-        : base(
-            Gtk.Internal.FileDialog.New(),
```
Comment (veler) [inline src/app/dev/platforms/desktop/DevToys.Linux/Core/FileStorage.cs:14]: THIS line is THE line that was making the app crashing on startup...

File: src/app/dev/platforms/desktop/DevToys.Linux/Core/FileStorage.cs [hunk 4]
```diff
@@ -69,47 +61,39 @@ public FileStream OpenWriteFile(string relativeOrAbsoluteFilePath, bool replaceI
...
-            else
-            {
-                tcs.SetResult(new Gio.FileHelper(fileValue, true));
-            }
-        });
+            taskCompletionSource.SetResult(folder);
+        };
 
-        Gtk.Internal.FileDialog.SelectFolder(
-            self: base.Handle,
```
Comment (veler) [inline src/app/dev/platforms/desktop/DevToys.Linux/Core/FileStorage.cs:103]: We needed `Gtk.Internal.FileDialog.New()` HERE. For THIS line. Now we use `Gtk.FileChooserNative.New`.

File: src/app/dev/platforms/desktop/DevToys.Linux/Core/ThemeListener.cs [hunk 5]
```diff
@@ -126,20 +140,14 @@ private bool GetBestValueForCompactMode()
 
     private void UpdateSystemSettingsAndApplyTheme()
     {
-        IsHighContrast = false; // TODO: Detect high contrast
+        IsHighContrast = _adwStyleManager.HighContrast;
```
Comment (veler) [inline src/app/dev/platforms/desktop/DevToys.Linux/Core/ThemeListener.cs:143]: We can now detect High Contrast, hooray!

File: src/app/dev/platforms/desktop/DevToys.MacOS/Controls/BlazorWebView/BlazorWKWebView.cs [hunk 6]
```diff
@@ -10,49 +10,48 @@ namespace DevToys.MacOS.Controls.BlazorWebView;
 // TODO: Credit MAUI repository
 internal sealed partial class BlazorWkWebView : IDisposable
 {
+    private const string DevToysInteropName = "devtoyswebinterop";
```
Comment (veler) [inline src/app/dev/platforms/desktop/DevToys.MacOS/Controls/BlazorWebView/BlazorWKWebView.cs:13]: This is a tiny refactoring here. When I tried to make my own GTK wrapper for Linux I had to recode the whole BlazorWebView and learned a bit more about it. In a separate PR, at the occasion, I will refactor both BlazorWebView from MacOS and Linux to have much more code in common.

**Review by btiteux — APPROVED** [review]

**Pull request merged**